src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    public MyLinkedList(T element) {
+        Node<T> firstNode = new Node<>(element);
+        head = tail = firstNode;
+        size = 1;
+    }
+
+    private static class Node<T1> {
+        T1 storedItem;
+        Node<T1> previous;
+        Node<T1> next;
+
+        public Node(T1 element) {
+            storedItem = element;
+            previous = null;
+            next = null;
+        }
+    }
+
+    public void checkBounds(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is out of LinkedList bonds");
+        }
+    }
+
+    public void nodeInsertion(Node<T> previousNode, Node<T> nodeToInsert, Node<T> nextNode) {
+        if (previousNode == null) {
+            nodeToInsert.next = nextNode;
+            nodeToInsert.previous = null;
+            nextNode.previous = nodeToInsert;
+            head = nodeToInsert;
+        } else {
+            nodeToInsert.next = nextNode;
+            nodeToInsert.previous = previousNode;
+            previousNode.next = nodeToInsert;
+            nextNode.previous = nodeToInsert;
+        }
+    }
+
+    public void removeOnIndex(int nodePosition, Node<T> nodeToRemove) {
+        if (size == 1) {
+            head = null;
+            tail = null;
+        } else if (nodePosition == 0) {
+            nodeToRemove.next.previous = null;
+            head = nodeToRemove.next;
+        } else if (nodePosition == size - 1) {
+            nodeToRemove.previous.next = null;
+            tail = nodeToRemove.previous;
+        } else {
+            nodeToRemove.previous.next = nodeToRemove.next;
+            nodeToRemove.next.previous = nodeToRemove.previous;
+        }
+        nodeToRemove.next = nodeToRemove.previous = null;
+    }
+
+    public Node<T> findNode(int index) {
+        Node<T> headHandler = head;
+        Node<T> tailHandler = tail;
+        if (index < size / 2) {
+            for (int i = 0; i < index && headHandler.next != null; i++) {
```

Comment:
```suggestion
            for (int i = 0; i < index; i++) {
```
do we need another check?

Reply:
done